sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -149,7 +149,13 @@ class CSVOptions(
     parameters.getOrElse(DateTimeUtils.TIMEZONE_OPTION, defaultTimeZoneId))
 
   // A language tag in IETF BCP 47 format
-  val locale: Locale = parameters.get(LOCALE).map(Locale.forLanguageTag).getOrElse(Locale.US)
+  val locale: Locale = parameters.get(LOCALE)
+    .map {
+      case null =>
```

Comment:
My concern is that, in PySpark we ignore when it's `None` vs here we throw an exception. I really think we should fix the whole thing with defining the behaviour instead of fixing one case alone.

Reply:
I see. Would it make more sense to just add a null check here then?https://github.com/apache/spark/blob/2516fd8439df42b1c161fbd346a0c346cc075f0f/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L108

However, I'm not sure if it would adversely affect other options.

Reply:
Yeah, that's the key. This is actually a bigger problem than this specific case. I think we should define the behaviour, document somewhere, and apply them for all...